### ISSUE TITLE
Training loss unconverged, output unmatched with input

### ISSUE THREAD

**Issue by shanemankiw** [lllyasviel/ControlNet#215]
Hi, thanks for this great work!
We try to train the model to inpaint, followed your instructions to generate prompts from BLIP and accumulate grad. Now we use a batchsize of 4 x grad accumulation of 16, but the loss just fluctuated over time and never went down. This is the loss curve:
![image](https://user-images.githubusercontent.com/34244000/222869418-d57211e8-65d6-419d-a2e0-932792012b6d.png)

And as a result, the output does not match the input part either. It has ok quality on its own, though.
I was considering, 
1. if I modify the DDPM pipeline, the output result(from noise, sample 50 DDIM steps) could be more aligned with input, do you have any suggestions? Also,
2. it still does not solve the training loss problem where time steps are randomly sampled.

**Comment by lllyasviel** [comment]
2.4k steps with 4x accumulation means 0.6k real steps. for sd 1.5 the converge usually happen at 4k to 6k real steps. in your case need to wait to 24k steps

**Comment by shanemankiw** [comment]
> 

Thanks for the quick reply! We will wait a while then!

**Comment by shanemankiw** [comment]
> 2.4k steps with 4x accumulation means 0.6k real steps. for sd 1.5 the converge usually happen at 4k to 6k real steps. in your case need to wait to 24k steps

Thanks! We will wait a while then.

**Comment by Jackqu** [comment]
Hi, Do you have good results

**Comment by shanemankiw** [comment]
> Hi, Do you have good results

Not really in this particular case. But I did discover the nature of diffusion model training, especially finetuning: the loss would fluctuate and sometimes would only go down slightly in log scale. However, it does not mean the network did not learn anything. The network might still be finetuned.

**Comment by Oguzhanercan** [comment]
@shanemankiw Hi, I am trying to finetune InstantID, which uses controlnet. And I get loss function like yours. If you get good results with controlnet can you share techniques and hyperparameters (optimizer related, batchsize, number of samples etc) ?

**Comment by shanemankiw** [comment]
> @shanemankiw Hi, I am trying to finetune InstantID, which uses controlnet. And I get loss function like yours. If you get good results with controlnet can you share techniques and hyperparameters (optimizer related, batchsize, number of samples etc) ?

For me the trick is to let it run a little longer... 
I did not adjust other hyperparameters in detail, but generally I found that the bigger the batchsize the better the performance

**Comment by Oguzhanercan** [comment]
> > @shanemankiw Hi, I am trying to finetune InstantID, which uses controlnet. And I get loss function like yours. If you get good results with controlnet can you share techniques and hyperparameters (optimizer related, batchsize, number of samples etc) ?
> 
> For me the trick is to let it run a little longer... I did not adjust other hyperparameters in detail, but generally I found that the bigger the batchsize the better the performance

Thanks,

**Comment by jiajiao272727** [comment]
May I ask how did you get those loss curves and loss data through training? I followed the process of the formal tutorial  ["Train a ControlNet to Control"](https://github.com/lllyasviel/ControlNet/blob/main/docs/train.md), but only get the model of the newest epoch in the "lightning_logs" folder and sample images in the "image_log" folder. Will it store all loss data in "loss_log.txt" like pix2pix? Thank you so much~
